napari_plugin_manager/_tests/test_qt_plugin_dialog.py

```diff
@@ -181,44 +176,16 @@ def set_blocked(self, plugin, blocked):
 
     widget = qt_plugin_dialog.QtPluginDialog()
     widget.show()
-    qtbot.wait(300)
+    qtbot.waitUntil(widget.isVisible, timeout=300)
     qtbot.add_widget(widget)
+    qtbot.waitUntil(lambda: widget.available_list.count() >= 2, timeout=300)
```

Comment:
Did we know how many entries we expect here? If the test ends before the thread yelding new plugin entries it may end with a crash whole test setup. We should wait until this thread ends working.

Reply:
Also lambdas are sometimes funky in these tests, so using a separate function might be slightly better

Reply:
Done :)